flink-core/src/main/java/org/apache/flink/api/connector/sink2/InitContext.java

```diff
@@ -35,16 +40,37 @@ public interface InitContext {
      */
     long INITIAL_CHECKPOINT_ID = 1;
 
-    /** @return The id of task where the committer is running. */
+    /**
+     * Get the id of task where the committer is running.
+     *
+     * @deprecated This method is deprecated since Flink 1.19. All metadata about the task should be
+     *     provided uniformly by {@link #getTaskInfo()}.
+     * @see <a
+     *     href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-382%3A+Unify+the+Provision+of+Diverse+Metadata+for+Context-like+APIs">
+     *     FLIP-382: Unify the Provision of Diverse Metadata for Context-like APIs </a>
+     */
     int getSubtaskId();
```

Comment:
1. Aren't we deprecating these methods?
2. Same for the default implementation.

Reply:
I have annotated these deprecated methods by `@Deprecated` in the `InitContext`.